### DISCUSSION TITLE
Fresh firefly install broke

### DISCUSSION THREAD

**Discussion by anastutemind** [firefly-iii/firefly-iii#4831] [Firefly III]
**Bug description**
I am running Firefly III version 5.5.12, and my problem is:

After installation when it generate the password and log in for the first time it says:

Unfortunately, this error was not recoverable :(. Firefly III broke. The error is:

FireflyIII\Support\Amount::getDefaultCurrencyByUser(): Return value must be of type FireflyIII\Models\TransactionCurrency, null returned

**Steps to reproduce**
Fresh new install of Firefly, generated password and tried loggin in for the first time

**Comment by JC5** [comment]
Did you run the migrations?

**Reply by anastutemind** [reply]
Okay, I got it working now with:
composer install
php artisan migrate:refresh --seed
php artisan firefly-iii:upgrade-database
php artisan passport:install
php artisan serve

and php extensions:
;extension=gd
;extension=intl
;extension=sodium
